dxr/src/values/ser_de.rs

```diff
@@ -233,11 +234,20 @@ pub(crate) mod value {
                         Ok(Value::array(value))
                     },
                     #[cfg(feature = "nil")]
-                    Field::Nil => Ok(Value::nil()),
+                    Field::Nil => {
+                        let _: IgnoredAny = map.next_value()?;
+                        Ok(Value::nil())
+                    },
                 }
             } else {
                 // <value></value>
-                Ok(Value::string(String::new()))
+                return Ok(Value::string(String::new()));
+            };
+
+            if let Some(key) = map.next_key::<Field>()? {
```

Comment:
Probably `Cow<str>` will be better here. Then the error could be more clear that you have unexpected key even if that is not a key that `Field` can recognize. Currently for the field `abcdef` you will get "unknown field" error instead of "Unexpected key" error, but I think, that this key in the first place is unexpected and only after that is unknown.

Also, it maybe worth to add some formal tests for the deserializer implementation using `serde_test`

Reply:
I'll look into this, thank you!